tests/promise/promise.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* globals describe, it, expect, CKEDITOR, window */
+
+'use strict';
+
+describe( 'Promise', function() {
```

Comment:
This test is... weird and IMO overcomplicated. Just check that the API exists and does the most basic thing (no timeouts).

Besides that I don't like `CKEDITOR.require` call inside the test function - it should be as simple as possible. I don't know the best practices of testing with AMD (and we should check them), but I would for instance move the require() call outside at least of `it()` if not of `describe()`.

BTW. I've stumbled upon http://rangle.io/blog/the-art-of-promise-based-architecture/ last week. The 2nd part of the video is great and touches testing (actually - one thing - that you can return a promise and Mocha will resolve it).